baybe/constraints/conditions.py

```diff
@@ -33,12 +36,31 @@ def _is_not_close(x: ArrayLike, y: ArrayLike, rtol: float, atol: float) -> np.nd
     return np.logical_not(np.isclose(x, y, rtol=rtol, atol=atol))
 
 
+def isclose(x: PolarsType, y: PolarsType, rtol: float, atol: float) -> Iterable[bool]:
+    """Return a boolean array indicating where ``x`` and ``y`` are close.
+
+    The equivalent to ``numpy.isclose``.
+
+    Args:
+        x: First input array to compare.
+        y: Second input array to compare.
+        rtol: The relative tolerance parameter.
+        atol: The absolute tolerance parameter.
+
+    Returns:
+        Returns a boolean array of where ``x`` and ``y`` are equal within the
+        given tolerances.
+
+    """
+    return ops.le(abs(x - y), atol + rtol * abs(y))
+
+
 # provide threshold operators
 _threshold_operators: dict[str, Callable] = {
     "<": ops.lt,
     "<=": ops.le,
-    "=": rpartial(np.isclose, rtol=0.0),
-    "==": rpartial(np.isclose, rtol=0.0),
+    "=": rpartial(isclose, rtol=0.0),
+    "==": rpartial(isclose, rtol=0.0),
     "!=": rpartial(_is_not_close, rtol=0.0),
     ">": ops.gt,
```

Comment:
shouldnt `_is_not_close` also be updated and use the new `isclose` as basis?

Reply:
You're right, however, after replacing it with our `isclose` it had same issues with mypy.

Reply:
i dont think that matters my suggestion is based on the logic that `_is_not_close` is kind of the negated version of `isclose`. Hence, it would be wrong to have a custom `isclose` but have `_is_not_close` use `isclose` from another package